common/djangoapps/entitlements/models.py

```diff
@@ -41,9 +41,13 @@ def get_days_until_expiration(self, entitlement):
         """
         Returns an integer of number of days until the entitlement expires
         """
-        expiry_date = entitlement.created + timedelta(self.expiration_period_days)
         now = datetime.now(tz=pytz.UTC)
-        return (expiry_date - now).days
+        if not entitlement.enrollment_course_run:
+            expiry_date = entitlement.created + timedelta(self.expiration_period_days)
+            return (expiry_date - now).days
+        else:
+            expiry_date = entitlement.created + timedelta(self.regain_period_days)
+            return (expiry_date - now).days
```

Comment:
This logic is not completely correct. You also need to account for course start.

Reply:
It should be the same logic that you had in `is_entitlement_regainable`.